components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/certificatechecktrigger/certificatechecktrigger.go

```diff
@@ -35,25 +34,22 @@ func (ss *CertificateCheck) Run(config models.Config) []models.CheckTriggerRespo
 	//This map will hold the Response against each IP
 	var finalResult []models.CheckTriggerResponse
 
-	certificate := config.Certificate
-
-	hostMap := configutils.GetNodeTypeMap(config.Hardware)
 	url := fmt.Sprintf("%s:%s%s", ss.host, ss.port, constants.CERTIFICATE_CHECK_API_PATH)
-	for _, node := range certificate.Nodes {
-		nodeTypes := hostMap[node.IP]
-		//construct the request for Certificate Check API
-		requestBody := models.CertificateCheckRequest{
-			RootCertificate:  certificate.RootCert,
-			PrivateKey:       node.Key,
-			NodeCertificate:  node.Cert,
-			AdminPrivateKey:  node.AdminKey,
-			AdminCertificate: node.AdminCert,
-		}
+	for _, certificate := range config.Certificate {
+		for _, node := range certificate.Nodes {
+			//construct the request for Certificate Check API
+			requestBody := models.CertificateCheckRequest{
+				RootCertificate:  node.RootCert,
```

Comment:
For Automate nodes, the root-ca won't be available, and the `/api/v1/checks/certificate` API returns an invalid request in this case, we may need to change the API implementation as well to handle this scenario.

Reply:
Added the nil case for root ca now. Now certificate API will except blank root ca.